### PR TITLE
SPUI-987 > Add dynamic gift bonus

### PR DESCRIPTION
**Pull request by DenisMartych**

### CODE REVIEW

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_header.dart [hunk 1]
```diff
@@ -83,4 +87,28 @@ class PortfolioWithBalanceHeader extends HookWidget {
       ),
     );
   }
+
+  String _giftBonus(List<CampaignOrReferralModel> rewards) {
+    var bonusGift = Decimal.zero;
+    for (final item in rewards) {
+      if (item.campaign != null) {
```
Comment (eli1stark) [inline lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_header.dart:94]: `if (item.campaign?.conditions?.isNotEmpty ?? false)`

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_header.dart [hunk 2]
```diff
@@ -83,4 +87,28 @@ class PortfolioWithBalanceHeader extends HookWidget {
...
+        if (item.campaign!.conditions != null &&
```
Comment (eli1stark) [inline lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_header.dart:95]: remove

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_header.dart [hunk 3]
```diff
@@ -83,4 +87,28 @@ class PortfolioWithBalanceHeader extends HookWidget {
       ),
     );
   }
+
+  String _giftBonus(List<CampaignOrReferralModel> rewards) {
+    var bonusGift = Decimal.zero;
```
Comment (eli1stark) [inline lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_header.dart:92]: add space

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_header.dart (hunk 1 above)
Comment (DenisMartych) [reply lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_header.dart:94]: done

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_header.dart (hunk 2 above)
Comment (DenisMartych) [reply lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_header.dart:95]: done

File: lib/app/screens/portfolio/view/components/portfolio_header.dart [hunk 4]
```diff
@@ -83,4 +91,26 @@ class PortfolioHeader extends HookWidget {
       ),
     );
   }
+
+  String _giftBonus(List<CampaignOrReferralModel> rewards) {
+    var bonusGift = Decimal.zero;
+
+    for (final item in rewards) {
+      if (item.campaign != null) {
```
Comment (eli1stark) [inline lib/app/screens/portfolio/view/components/portfolio_header.dart:99]: remove, redundant

File: lib/app/screens/portfolio/view/components/portfolio_header.dart [hunk 5]
```diff
@@ -64,11 +68,15 @@ class PortfolioHeader extends HookWidget {
                     children: [
                       const SGiftPortfolioIcon(),
                       if (gift == ReferralGiftStatus.showGift) ...[
-                        const SpaceW8(),
-                        Text(
-                          '\$15',
-                          style: sSubtitle3Style.copyWith(
-                            color: colors.white,
+                        Container(
+                          margin: (_giftBonus(rewards) != '')
```
Comment (eli1stark) [inline lib/app/screens/portfolio/view/components/portfolio_header.dart:72]: use `_giftBonus(rewards).isNotEmpty` instead

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/app/screens/portfolio/view/components/portfolio_header.dart (hunk 5 above)
Comment (DenisMartych) [reply lib/app/screens/portfolio/view/components/portfolio_header.dart:72]: done

File: lib/app/screens/portfolio/view/components/portfolio_header.dart (hunk 4 above)
Comment (DenisMartych) [reply lib/app/screens/portfolio/view/components/portfolio_header.dart]: done

**Pull request merged**